### PR TITLE
MINIFI-417 Upgrading underlying NiFi libs to 1.4.0 from 1.2.0

### PR DESCRIPTION
**Pull request by JPercivall**
Thank you for submitting a contribution to Apache NiFi - MiNiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi-minifi folder?
- [X] Have you written or updated unit tests to verify your changes?
- [X] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [X] If applicable, have you updated the LICENSE file, including the main LICENSE file under minifi-assembly?
- [X] If applicable, have you updated the NOTICE file, including the main NOTICE file found under minifi-assembly?

### For documentation related changes:
- [X] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by JPercivall** [comment]
Took a swag at updating the underlying libs from 1.2.0 to 1.4.0 and it didn't appear to be too bad. I built everything and updated anything that needed to be updated. I also checked the nifi.properties to see if we needed to add anything and in 1.3.0 and 1.4.0 the only changes were related to clustering or authorizers (not applicable).

I admittedly did a not good job testing the full suite of functionality. The tests that I ran were the full build, a basic S2S, and transformation using the toolkit. I've had a bad habit recently of starting things and if I don't commit/put up a PR in one go then it gets lost. So I just wanted to get something up. If updating to 1.4.0 this way is something we want to pursue, I'd appreciate any testing that people are able to do easily (ie. those that have a better understanding of the C2 and have an instance set-up).

File: minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java [hunk 1]
```diff
@@ -77,7 +77,7 @@
 
 public final class ConfigTransformer {
     // Underlying version of NIFI will be using
-    public static final String NIFI_VERSION = "1.2.0";
+    public static final String NIFI_VERSION = "1.4.0";
```
Comment (jzonthemtn) [inline minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java]: Any way this value could reference the `org.apache.nifi.version` value in the pom?

Comment (JPercivall) [reply minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java]: There appear to be a couple of different ways[1] but most of them are pretty hacky and not sure it's worth putting in for this one variable. If the variable was logic that was relied on for MiNiFi to work I'd probably lean towards adding it but it's currently only used in one place[2] and that is just for documentation within a file that in general isn't viewed by users (the underlying nifi.properties).

[1] https://stackoverflow.com/questions/11500533/access-maven-properties-defined-in-the-pom
[2] https://github.com/JPercivall/nifi-minifi/blob/baac89b1ddd7a019635bd150c0cce9a37314a371/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java#L163

**Comment by apiri** [comment]
reviewing

**Comment by apiri** [comment]
code looks fine and was able to run it through some flows/configurations that were highlighting various problem spots addressed with 0.3.0.  Will get this merged in.  Thanks @JPercivall!

**Comment by JPercivall** [comment]
Awesome, thanks @apiri. I'm glad this wasn't as bad as the upgrade you did for 1.2.0

**Pull request closed without merging**